guides/guide-for-adding-windows-node.md

```diff
@@ -23,148 +18,11 @@ Your Kubernetes server must be at or later than version 1.22. To check the versi
-```
-
-
-## Configuring Calico hostprocess
-
->  **Note:** All code snippets in Linux sections are to be run in a Linux environment on the Linux worker node.
-
-1. Prepare Kubernetes control plane for Calico
-
-Some minor preparation is recommended on the Kubernetes control plane in our cluster. It is recommended to enable bridged IPv4 traffic to iptables chains when using Calico. The following command must be run on all Linux nodes:
-
-```bash
-sudo sysctl net.bridge.bridge-nf-call-iptables=1
-```
-
-2. Download & configure Calico for Linux
-
-The version of Calico will be set in the variable CALICO_VERSION (if another version is needed, modify this variable).
-
-```bash
-export CALICO_VERSION="v3.24.5"
-```
-
-Create the tigera-operator.yaml and the custom-resources.yaml for Calico: 
-```bash
-kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/$CALICO_VERSION/manifests/tigera-operator.yaml
-curl https://raw.githubusercontent.com/projectcalico/calico/$CALICO_VERSION/manifests/custom-resources.yaml -O
-# If IP CIDR differs from 192.168.0.0/16, make sure to modify it in custom-resources.yaml.
-# After the file has the correct CIDR, run this command:
-kubectl create -f custom-resources.yaml
-```
->  **Note:** Before creating this manifest, read its contents and make sure its settings are correct for your environment. For example, you may need to change the default IP pool CIDR to match your pod network CIDR.
-
-Remove the taints on the master so that you can schedule pods on it.
-
-```bash
-kubectl taint nodes --all node-role.kubernetes.io/control-plane-
-```
-
-It should return the following.
-
-```
-node/<your-hostname> untainted
-```
-3. Add calico and kube-proxy daemonsets from sig-windows-tools/hostprocess/calico
-
-Now you can add Windows-compatible versions of calico and kube-proxy. In order to ensure that you get a compatible version of kube-proxy, you'll need to substitute the tag of the image. The following example shows usage for Kubernetes v1.25.3, but you should adjust the version for your own deployment.
-
-```bash
-curl -L https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/master/hostprocess/calico/kube-proxy/kube-proxy.yml | sed 's/KUBE_PROXY_VERSION/v1.25.3/g' | kubectl apply -f -
-curl -L https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/master/hostprocess/calico/calico.yml | sed 's/CALICO_VERSION/$CALICO_VERSION/g' | kubectl apply -f -
-kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/master/hostprocess/calico/kube-calico-rbac.yml
-```
-
->  **Note** To find your version of kubernetes run the following command:
-> `kubeadm version`
-
 ### Joining a Windows worker node
 
 > **Note:** All code snippets in Windows sections are to be run in a PowerShell environment with elevated permissions (Administrator) on the Windows worker node.
 
-1. Install ContainerD, wins, kubelet, and kubeadm.
+1. Install ContainerD, kubelet, and kubeadm.
```

Comment:
nit: add Git installation step before `.\PrepareNode.ps1` step since it's depended on it.

```
C:\Users\Administrator\k8s-setup\PrepareNode.ps1 : The term 'git' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is
correct and try again.
At line:1 char:1
+ .\PrepareNode.ps1 -KubernetesVersion v1.26.0
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
nit: make `.\PrepareNode.ps1` script idempotent; fails after running the second time when an initially created file is re-created:
```
----                 -------------         ------ ----
d-----         1/22/2023  10:28 PM                pki
New-Item : NewItemIOError
At C:\Users\Administrator\k8s-setup\PrepareNode.ps1:61 char:1
+ New-Item -path C:\var\lib\kubelet\etc\kubernetes\pki -type SymbolicLi ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceExists: (C:\var\lib\kubelet\etc\kubernetes\pki:String) [New-Item], IOException
    + FullyQualifiedErrorId : NewItemIOError,Microsoft.PowerShell.Commands.NewItemCommand
```

Actually it's only these 2 other places:
```
Registering kubelet service
Error creating service!
CreateService(): The specified service already exists.
```
----
```
Set parameter "DependOnService" for service "kubelet".
New-NetFirewallRule : Cannot create a file when that file already exists.
At C:\Users\Administrator\k8s-setup\PrepareNode.ps1:106 char:1
+ New-NetFirewallRule -Name kubelet -DisplayName 'kubelet' -Enabled Tru ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceExists: (MSFT_NetFirewallRule:root/standardcimv2/MSFT_NetFirewallRule) [New-NetF
   irewallRule], CimException
    + FullyQualifiedErrorId : Windows System Error 183,New-NetFirewallRule
```

Reply:
> nit: add Git installation step before `.\PrepareNode.ps1` step since it's depended on it.

https://github.com/kubernetes-sigs/sig-windows-tools/pull/280 should resolve this

> nit: make .\PrepareNode.ps1 script idempotent; fails after running the second time when an initially created file is re-created:

let's do a follow up for this. This PR is just a copy/paste so it is easier to maintain